pkgs/os-specific/linux/kernel/generic.nix

```diff
@@ -117,6 +118,8 @@ let
         map ({extraConfig ? "", ...}: extraConfig) kernelPatches;
     in lib.concatStringsSep "\n" ([baseConfigStr] ++ configFromPatches);
 
+  withRust = ((configfile.moduleStructuredConfig.settings.RUST or {}).tristate or null) == "y";
```

Comment:
If Rust support is ever opt-out (i.e. on-by-default), then *that* is where you would put the `isX86` check -- i.e. `, withRust ? stdenv.hostPlatform.isx86`.